src/components/dbCalls.js

```diff
@@ -0,0 +1,435 @@
+        const newAddition = {
+            time: time,
+            log: newLog
+        }
+        const newLogs = [...currLogs, newAddition];
+        await updateDoc(docRef, { logs: newLogs });
+        return newLogs;
+    }
+    catch (error) {
+        console.error("Error updating logList: ", error);
+    }
+}
+
+//fetches a player's targets from database
+const fetchTargetsForPlayer = async (playerName, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', playerName));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerTargets = playerSnapshot.docs[0].data().targets;
+        return playerTargets;
+    }    
+    catch (error) {
+        console.error('Error fetching player targets: ', error);
+    }
+}
+
+//fetches all active tasks from database
+const fetchActiveTasksForRoom = async (roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskQuery = query(taskCollectionRef, where ('isComplete', '==', false));
+        const taskSnapshot = await getDocs(taskQuery);
+        return taskSnapshot;
+    }
+    catch (error) {
+        console.error('Error fetching active tasks: ', error);
+    }
+}
+
+//fetches all inactive tasks from database
+const fetchInactiveTasksForRoom = async (roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskQuery = query(taskCollectionRef, where ('isComplete', '==', true));
+        const taskSnapshot = await getDocs(taskQuery);
+        return taskSnapshot;
+    }
+    catch (error) { 
+        console.error('Error fetching inactive tasks: ', error);
+    }
+}
+
+//fetches a task's data from database
+const fetchTaskForRoom = async (taskID, roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskRef = doc(taskCollectionRef, taskID);
+        const taskSnapshot = await getDoc(taskRef);
+        return taskSnapshot.data();
+    }
+    catch (error) {
+        console.error('Error fetching task: ', error);
+    }
+}
+
+//updates player's score
+const updatePointsForPlayer = async (player, points, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        const playerScore = parseInt(playerSnapshot.docs[0].data().score);
+        const newPoints = points + playerScore;
+        await updateDoc(playerdoc, { score: newPoints });
+    }
+    catch (error) {
+        console.error('Error updating player score: ', error);
+    }
+}
+
+//revives a player
+const updateIsAliveToTrueForPlayer = async (player, roomID) => {
```

Comment:
You can combine this with the below method and take in "isAlive" as a parameter. That'll simplify your code a good bit